pkgs/development/libraries/gnu-efi/default.nix

```diff
@@ -34,7 +34,9 @@ stdenv.mkDerivation rec {
     description = "GNU EFI development toolchain";
     homepage = "https://sourceforge.net/projects/gnu-efi/";
     license = licenses.bsd3;
-    platforms = platforms.linux;
+    platforms = with lib.systems.inspect.patterns; map (pat: pat // isLinux) ([
+      isArmv6 ] ++ isArmv7 ++ [ isArmv8 isRiscV isx86
+    ]);
```

Comment:
Using doubles strings rather than patterns would make the code here less complicated, and have the same effect, right?

Reply:
I thought @Ericson2314 wanted to get rid of double strings but maybe I misinterpreted him.

Reply:
It's not something we're going to move away from one package at a time regardless…

Reply:
> Using doubles strings rather than patterns would make the code here less complicated, and have the same effect, right?

Not exactly.

The `isEfi` predicate that I removed uses a pattern, so to get precisely the same effect I used a pattern here.  The main issue is that `isArmv7` is defined to be *any* CPU-string which has the prefix `armv7`.  And there are a lot of those!

Also double-strings are icky gross.